### ISSUE TITLE
Binder is not accessible from Microsoft Edge (works perfectly fine in Mozilla Firefox)

### ISSUE THREAD

**Issue by kcostya** [jupyterhub/mybinder.org-deploy#241]
Binder is not accessible from Microsoft Edge but works perfectly fine in Mozilla Firefox:
https://mybinder.org/v2/gh/jvns/pandas-cookbook/master

In Edge the page stays at the following, and 'launch' button just refreshes it:
![image](https://user-images.githubusercontent.com/34856477/34354075-81752bbc-ea34-11e7-90fb-6f2197933623.png)

**Comment by choldgraf** [comment]
I can confirm this happens to me as well on Edge. I'm getting an `SCRIPT5009: 'EventSource' is not defined` error in the console when I click launch!

**Comment by willingc** [comment]
I believe that this is happening since Edge does not support Server Sent Events (SSE) [ref](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events). We should look into the possibility of a workaround for Edge. In the meanwhile, please use another browser. Thanks for the report @kcostya

**Comment by choldgraf** [comment]
Looks like this might be a possible workaround from our end: https://github.com/Yaffle/EventSource

That said, just a note that ~2-3% of our traffic is coming from IE/Edge, so we could also do a check for those browsers and display a message like "please use either Firefox or Chrome to use Binder"

![image](https://user-images.githubusercontent.com/1839645/34379889-f07a4b0e-eafd-11e7-8359-f3951483c6d6.png)

**Comment by betatim** [comment]
I had a look at the repo you linked. It looks like what they do is send 2kB of data on the event stream and that is "it"? In which case we could do that in tornado too right?
```
...
    var padding = new Array(2049);
    response.write(":" + padding.join(" ") + "\n"); // 2kB padding for IE
    response.write("retry: 2000\n");
...
```

**Comment by yuvipanda** [comment]
@betatim it needs both the extra padding in tornado, and a client side shim (since the EventSource object/class isn't available in Edge).

**Comment by choldgraf** [comment]
In the meantime, what if we added a supported browsers section to the documentation? So on the FAQ page we could have:

## What browsers are supported with mybinder.org?

BinderHub is designed to work with recent versions of Mozilla Firefox and Google Chrome. These are both freely available on most modern operating systems. For other browsers such as Microsoft Internet Explorer or Edge, we do not guarantee 100% functionality.

**Comment by betatim** [comment]
Instead of saying "we promise nothing" say "we know it doesn't work for Edge (see [issue](https://github.com/jupyterhub/mybinder.org-deploy/issues/241) for discussion and how it could be fixed, PRs welcome)"

Maybe the bit in () needs to be a bit shorter and yet be a hook to encourage someone who cares about Edge compatibility to help us fix it ;)

**Comment by willingc** [comment]
Perhaps:

mybinder.org is designed to work with modern web browsers. At present, Firefox, Chrome, and Safari are supported. While we would like to support Microsoft Edge, Edge does not currently implement a key function, Server Sent Events (SSE). We welcome PRs to improve Edge support. If you are using Edge or an older browser (i.e. Internet Explorer), please use another browser for a better user experience.

**Comment by choldgraf** [comment]
@willingc I like that wording as well...shall I make a PR?

**Comment by yuvipanda** [comment]
Now that we use npm, I think we can fix this by adding https://www.npmjs.com/package/event-source-polyfill to our package.json and importing it for our EventSource object. That should definitely support Edge, and I think I'm happy for us to not support older IE.

**Comment by choldgraf** [comment]
I opened https://github.com/jupyterhub/binderhub/pull/407 ...

**Comment by choldgraf** [comment]
ok, I believe this is fixed for Edge by #407 - @kcostya want to give it another shot and we can close this if all looks well for you?

**Comment by kcostya** [comment]
@choldgraf thanks a lot!
The issue is resolved.
